test/videojs.ads.test.js

```diff
@@ -1,1025 +1,732 @@
+  var className = this.player.el().className;
+  assert.ok(className.indexOf('vjs-ad-playing') >= 0, 'the ad class should be in "' + className + '"');
+});
+
+QUnit.test('removes the ad-mode class when a preroll finishes', function(assert) {
+  assert.expect(5);
+  this.player.trigger('adsready');
+  this.player.tech.trigger('play');
+  this.player.ads.startLinearAdMode();
+  this.player.ads.endLinearAdMode();
+  var className = this.player.el().className;
+  assert.ok(className.indexOf('vjs-ad-playing') < 0, 'the ad class should not be in "' + className + '"');
+  assert.strictEqual(this.contentPlaybackSpy.callCount, 0, 'did not fire contentplayback yet');
+  assert.strictEqual(this.player.ads.triggerevent, 'adend', 'triggerevent for content-resuming should have been adend');
+  this.player.trigger('playing');
+  assert.strictEqual(this.contentPlaybackSpy.callCount, 1, 'A contentplayback event should have triggered');
+  assert.strictEqual(this.contentPlaybackReason(), 'playing', 'The triggerevent for content-playback should have been playing');
+});
+
+QUnit.test('adds a class while waiting for an ad plugin to load', function(assert) {
+  assert.expect(1);
+  this.player.tech.trigger('play');
+  var className = this.player.el().className;
+  assert.ok(className.indexOf('vjs-ad-loading') >= 0, 'the ad loading class should be in "' + className + '"');
+});
+
+QUnit.test('adds a class while waiting for a preroll', function(assert) {
+  assert.expect(1);
+  this.player.trigger('adsready');
+  this.player.tech.trigger('play');
+  var className = this.player.el().className;
+  assert.ok(className.indexOf('vjs-ad-loading') >= 0, 'the ad loading class should be in "' + className + '"');
+});
+
+QUnit.test('removes the loading class when the preroll begins', function(assert) {
+  assert.expect(1);
+  this.player.trigger('adsready');
+  this.player.tech.trigger('play');
+  this.player.ads.startLinearAdMode();
+  var className = this.player.el().className;
+  assert.ok(className.indexOf('vjs-ad-loading') < 0, 'there should be no ad loading class present in "' + className + '"');
+});
+
+QUnit.test('removes the loading class when the preroll times out', function(assert) {
+  assert.expect(3);
+  this.player.trigger('adsready');
+  this.player.tech.trigger('play');
+  this.player.trigger('adtimeout');
+  var className = this.player.el().className;
+  assert.ok(className.indexOf('vjs-ad-loading') < 0, 'there should be no ad loading class present in "' + className + '"');
+  assert.strictEqual(this.contentPlaybackSpy.callCount, 1, 'A content-playback event should have triggered');
+  assert.strictEqual(this.contentPlaybackReason(), 'adtimeout', 'The reason for content-playback should have been adtimeout');
+});
+
+QUnit.test('starts the content video if there is no preroll', function(assert) {
+  assert.expect(3);
+  var spy = sinon.spy(this.player, 'play');
+  this.player.trigger('adsready');
+  this.player.tech.trigger('play');
+  this.clock.tick(1);
```

Comment:
I'm not sure, but it exists because we need to wait for the `player.ads.cancelPlayTimeout` (which is a `defer`, i.e. 1ms) to resolve.